apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerDataReader.cs

```diff
@@ -0,0 +1,349 @@
+            return (int) _innerList[i].ConvertToClrType(GetSpannerFieldType(i), typeof(int));
+        }
+
+        /// <inheritdoc />
+        public override long GetInt64(int i)
+        {
+            return (long) _innerList[i].ConvertToClrType(GetSpannerFieldType(i), typeof(long));
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="i"></param>
+        /// <returns></returns>
+        public Value GetJsonValue(int i)
+        {
+            return (Value) _innerList[i].ConvertToClrType(GetSpannerFieldType(i), typeof(Value));
+        }
+
+        /// <inheritdoc />
+        public override string GetName(int i)
+        {
+            return _resultset.GetMetadataAsync(CancellationToken.None).Result.RowType.Fields[i].Name;
+        }
+
+        /// <inheritdoc />
+        public override int GetOrdinal(string name)
+        {
+            GaxPreconditions.CheckNotNullOrEmpty(name, nameof(name));
+            var fields = _resultset.GetMetadataAsync(CancellationToken.None).Result.RowType.Fields;
+            for (var i = 0; i < fields.Count; i++)
+                if (Compare(name, fields[i].Name, StringComparison.Ordinal) == 0)
+                    return i;
+            return -1;
+        }
+
+        /// <inheritdoc />
+        public override string GetString(int i)
+        {
+            return (string) _innerList[i].ConvertToClrType(GetSpannerFieldType(i), typeof(string));
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="i"></param>
+        /// <returns></returns>
+        public Timestamp GetTimestamp(int i)
+        {
+            return (Timestamp) _innerList[i].ConvertToClrType(GetSpannerFieldType(i), typeof(Timestamp));
+        }
+
+        /// <inheritdoc />
+        public override object GetValue(int i)
+        {
+            return this[i];
+        }
+
+        /// <inheritdoc />
+        public override int GetValues(object[] values)
+        {
+            var count = Math.Min(FieldCount, values.Length);
+            for (var i = 0; i < count; i++)
+                values[i] = this[i];
+            return count;
+        }
+
+
+        /// <inheritdoc />
+        public override bool IsDBNull(int i)
+        {
+            return GetJsonValue(i).KindCase == Value.KindOneofCase.NullValue;
+        }
+
+        /// <inheritdoc />
+        public override bool NextResult()
+        {
+            throw new NotSupportedException("Spanner does not support multiple SQL queries in a single command.");
+        }
+
+        /// <inheritdoc />
+        public override bool Read()
+        {
+            return ReadAsync(CancellationToken.None).Result;
+        }
+
+        /// <inheritdoc />
+        public override Task<bool> ReadAsync(CancellationToken cancellationToken)
+        {
+            return ExecuteHelper.WithErrorTranslationAndProfiling(async () =>
+            {
+                await GetMetadataAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
I've renamed it to populatemetadataasync -- but kept the task<resultsetmetadata> as its just way too convenient in a lot of places to get that as a result.  I think the new name and return type should be enough to communicate what its doing...